packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2430,57 +2673,67 @@ class ObjectsInRedisClient {
                     return tools.maybeCallbackWithError(callback, ERRORS.ERROR_DB_CLOSED);
                 }
 
-                let objects;
+                if (!keys) {
+                    return tools.maybeCallbackWithError(callback, ERRORS.ERROR_NOT_FOUND);
+                }
+
+                let strObjects;
                 try {
-                    objects = await this.client.mget(keys);
+                    strObjects = await this.client.mget(keys);
                 } catch (e) {
                     return tools.maybeCallbackWithRedisError(callback, e);
                 }
                 const filteredKeys = [];
                 const filteredObjs = [];
-                objects = objects || [];
-                for (let k = 0; k < keys.length; k++) {
+
+                for (let i = 0; i < keys.length; i++) {
+                    const strObj = strObjects[i];
+                    let obj: null | ioBroker.Object;
                     try {
-                        objects[k] = JSON.parse(objects[k]);
+                        obj = strObj ? JSON.parse(strObj) : null;
                     } catch {
-                        this.log.error(`${this.namespace} Cannot parse JSON ${keys[k]}: ${objects[k]}`);
+                        this.log.error(`${this.namespace} Cannot parse JSON ${keys[i]}: ${strObj}`);
                         continue;
                     }
-                    if (!utils.checkObject(objects[k], options, CONSTS.ACCESS_WRITE)) {
+                    if (!obj || !utils.checkObject(obj, options, CONSTS.ACCESS_WRITE)) {
                         continue;
                     }
-                    if (!objects[k].acl) {
-                        objects[k].acl = {
+                    if (!obj.acl) {
+                        obj.acl = {
                             owner: (this.defaultNewAcl && this.defaultNewAcl.owner) || CONSTS.SYSTEM_ADMIN_USER,
                             ownerGroup:
                                 (this.defaultNewAcl && this.defaultNewAcl.ownerGroup) || CONSTS.SYSTEM_ADMIN_GROUP,
                             object:
                                 (this.defaultNewAcl && this.defaultNewAcl.object) ||
                                 CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ // '0644'
                         };
-                        if (objects[k].type === 'state') {
-                            objects[k].acl.state =
+                        if (obj.type === 'state') {
+                            // @ts-expect-error we have ensured above, that property acl exists
+                            obj.acl.state =
```

Comment:
I think the real issue here is the order of narrowing. I'm not sure wtf TS is doing here, you could rewrite the entire assignment though:
```ts
                        obj.acl = {
                            owner: (this.defaultNewAcl && this.defaultNewAcl.owner) || CONSTS.SYSTEM_ADMIN_USER,
                            ownerGroup:
                                (this.defaultNewAcl && this.defaultNewAcl.ownerGroup) || CONSTS.SYSTEM_ADMIN_GROUP,
                            object:
                                (this.defaultNewAcl && this.defaultNewAcl.object) ||
                                CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ, // '0644'
                            state:
                                obj.type === 'state'
                                    ? (this.defaultNewAcl && this.defaultNewAcl.state) ||
                                      CONSTS.ACCESS_USER_RW | CONSTS.ACCESS_GROUP_READ | CONSTS.ACCESS_EVERY_READ // '0644'
                                    : undefined
                        } as ioBroker.ObjectACL | ioBroker.StateACL;
```

Alternatively:
```suggestion:
obj.acl!.state = 
```
and keep the rest as is